paper/forestecology.Rmd

```diff
@@ -0,0 +1,456 @@
+comp_bayes_lm_scbi <- focal_vs_comp_scbi %>%
+  comp_bayes_lm(prior_param = NULL)
+```
+
+The returned `comp_bayes_lm_scbi` output is an object of S3 class type `comp_bayes_lm` which contains the posterior values of all parameters in our competition Bayesian linear regression. This class of object includes generic methods implemented for `print()`, `predict()`, and `ggplot2::autoplot()`. First the generic for `print()` displays the names of all prior & posterior parameters along with the model formula:
+
+```{r}
+comp_bayes_lm_scbi
+```
+
+Next, the generic for `predict()` takes as inputs the posterior parameter values in `comp_bayes_lm_scbi` and the predictor variables in `newdata` and outputs a vector of fitted/predicted values $\widehat{y}$ of the DBH for each focal tree computed from the posterior predictive distribution.
+
+```{r scbi-model-predict}
+focal_vs_comp_scbi <- focal_vs_comp_scbi %>%
+  mutate(growth_hat = predict(comp_bayes_lm_scbi, newdata = focal_vs_comp_scbi))
+```
+```{r}
+focal_vs_comp_scbi
+```
+
+We then compare the observed and fitted/predicted growths to compute the root mean squared error (RMSE) of our model fit.
+
+```{r scbi-model-rmse}
+model_rmse <- focal_vs_comp_scbi %>%
+  rmse(truth = growth, estimate = growth_hat) %>%
+  pull(.estimate)
+model_rmse
+```
+
+Lastly, the generic for `ggplot2::autoplot()` allows us to plot the posterior distribution of all parameters in Figure \ref{fig:scbi-posterior-viz} (for compactness we only show posteriors for 3 species). 
+
+```{r scbi-posterior-viz, out.width="100%", fig.cap= "Fit model: Posterior distributions of all parameters for three species."}
+# Plot posteriors for only a subset of species
+sp_to_plot <- c("litu", "quru", "cagl")
+
+plot1 <- autoplot(comp_bayes_lm_scbi, type = "intercepts", sp_to_plot = sp_to_plot)
+plot2 <- autoplot(comp_bayes_lm_scbi, type = "dbh_slopes", sp_to_plot = sp_to_plot)
+plot3 <- autoplot(comp_bayes_lm_scbi, type = "competition", sp_to_plot = sp_to_plot)
+
+# Combine plots using patchwork
+(plot1 | plot2) / plot3
+```
+
+These plots give the posterior distributions of parameters from Equation \ref{eq:model}. For many pacakge users they will be of interest because they give insight into the species-specific competitive interactions. Setting `type = "intercepts"` gives posterior distributions for $\beta_{0,j}$ and `type = "dbh_slopes"` for $\beta_{dbh,j}$. These give species specific growth independent of competition. The values of more interest are plotted with `type = 'competition'` which gives the posterior distribution for $\lambda_{j,k}$ species-specific competition coefficients (i.e., the $\lambda$-matrix). Negative values indicate a competitor species which slows the growth of a focal species. Here, for example, we see that tulip trees (`litu`) have a strong negative effect on the growth of conspecifics but relatively little effect on neighbors of the other two species.
```

Comment:
Love the way you laid out this introduction to model fitting. Bravo.☃️

Reply:
This is all @dallenmidd!